<launch>
  <arg name="uuv_name" />
  <arg name="start_time" default="-1"/>
  <arg name="radius" default="8"/>
  <arg name="center_x" default="0"/>
  <arg name="center_y" default="0"/>
  <arg name="center_z" default="-30"/>
  <arg name="n_points" default="50"/>
  <arg name="n_turns" default="1"/>
  <arg name="delta_z" default="5.0"/>
  <!-- Heading offset given in degrees -->
  <arg name="heading_offset" default="0"/>
  <arg name="duration" default="150"/>
  <arg name="max_forward_speed" default="0.3"/>

  <group>
    <push-ros-namespace namespace="$(var uuv_name)"/>
    <node pkg="uuv_control_utils"
      exec="start_helical_trajectory.py"
      name="start_helical_trajectory"
      output="screen">
      <param name="start_time" value="$(var start_time)"/>
      <param name="radius" value="$(var radius)"/>
      <param name="center" value="$(var center_x), $(var center_y), $(var center_z)" value-sep=", "/>
      <param name="n_points" value="$(var n_points)"/>
      <param name="heading_offset" value="$(var heading_offset)"/>
      <param name="max_forward_speed" value="$(var max_forward_speed)"/>
      <param name="duration" value="$(var duration)"/>
      <param name="n_turns" value="$(var n_turns)"/>
      <param name="delta_z" value="$(var delta_z)"/>
    </node>
  </group>
</launch>
